cnf-certification-test/lifecycle/podsets/podsets.go

```diff
@@ -37,7 +37,9 @@ var WaitForDeploymentSetReady = func(ns, name string, timeout time.Duration) boo
 	start := time.Now()
 	for time.Since(start) < timeout {
 		dp, err := provider.GetUpdatedDeployment(clients.K8sClient.AppsV1(), ns, name)
-		if err == nil && IsDeploymentReady(dp) {
+		if dp == nil {
```

Comment:
Sorry, Brandon, but I still don't see the need for this check `dp == nil`. Usually, after a function call in go, the first thing to check is the error with a `if err != nil`, and that's precisely the only moment where dp will be nil, so there's no need for another check for that. I suggest this way of checking the output:
```
	for time.Since(start) < timeout {
		dp, err := provider.GetUpdatedDeployment(clients.K8sClient.AppsV1(), ns, name)
		if err != nil {
			logrus.Errorf("Error while getting deployment %s (ns: %s), err: %s", name, ns, err)
		} else if !IsDeploymentReady(dp) {
			logrus.Errorf("%s is not ready yet", provider.DeploymentToString(dp))
		} else {
			logrus.Tracef("%s is ready!", provider.DeploymentToString(dp))
			return true
		}
	    
		time.Sleep(time.Second)
	}
```

Reply:
Okay, I agree this works.  I will update.